### PR TITLE
ffmpeg: some more improvements

### PR DESCRIPTION
**Pull request by jopejoe1**
## Description of changes

- set pkgconfig modules depending on library
- enable opengl and rtmp
- add a comment about tensorflow being disabled
- ~point changelog to the src.rev~ tested this wrongly
- make implicit dependency on nvcodec explicit
- give cuvid its own option
- ~clean up the darwin libraries~ moving this into separated pr

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by jopejoe1** [comment]
Moved the darwin stuff out of this pr will create a separated pr for that

File: pkgs/development/libraries/ffmpeg/generic.nix [hunk 1]
```diff
@@ -91,7 +91,7 @@
 , withSsh ? withHeadlessDeps # SFTP protocol
 , withSvg ? withFullDeps # SVG protocol
 , withSvtav1 ? withHeadlessDeps && !stdenv.isAarch64 && !stdenv.hostPlatform.isMinGW # AV1 encoder/decoder (focused on speed and correctness)
-, withTensorflow ? false # Tensorflow dnn backend support
+, withTensorflow ? false # Tensorflow dnn backend support # Increases closure size by ~390 MiB
```
Comment (Atemu) [inline pkgs/development/libraries/ffmpeg/generic.nix:96]: If it does indeed eval without allowInsecure now and "only" adds 390MiB, we could think about enabling it.

I'd like to resolve https://github.com/NixOS/nixpkgs/issues/271863 beforehand though.

**Review by Atemu — COMMENTED** [review]
Did you test the newly enabled features?

***

> [opengpl](https://github.com/NixOS/nixpkgs/pull/292498/commits/b7c1accda3b74dbd942a180c5f7c17a401d105b8)

The crossover I never knew I wanted :p

File: pkgs/development/libraries/ffmpeg/generic.nix [hunk 2]
```diff
@@ -27,20 +27,26 @@
...
-, withCuda ? withFullDeps && (with stdenv; (!isDarwin && !hostPlatform.isAarch && !hostPlatform.isRiscV))
+, withCoreImage ? withHeadlessDeps && stdenv.isDarwin # Apple CoreImage framework
+, withCuda ? withFullDeps && withFfnvcodec
 , withCudaLLVM ? withFullDeps
+, withCuvid ? withHeadlessDeps && withFfnvcodec
 , withDav1d ? withHeadlessDeps # AV1 decoder (focused on speed and correctness)
 , withDc1394 ? withFullDeps && !stdenv.isDarwin # IIDC-1394 grabbing (ieee 1394)
 , withDrm ? withHeadlessDeps && (with stdenv; isLinux || isFreeBSD) # libdrm support
 , withFdkAac ? withFullDeps && (!withGPL || withUnfree) # Fraunhofer FDK AAC de/encoder
+, withFfnvcodec ? withHeadlessDeps && (with stdenv; !isDarwin && !isAarch32 && !hostPlatform.isRiscV && hostPlatform == buildPlatform) # dynamically linked Nvidia code
```
[the remark this answers is not in the document]
Comment (jopejoe1) [reply pkgs/development/libraries/ffmpeg/generic.nix]: Renamed

**Comment by jopejoe1** [comment]
> Did you test the newly enabled features?

Some basic testing

> > [opengpl](https://github.com/NixOS/nixpkgs/pull/292498/commits/b7c1accda3b74dbd942a180c5f7c17a401d105b8)
> 
> The crossover I never knew I wanted :p

 :p

**Pull request closed without merging**